tests/blob/comparison.sy

```diff
@@ -0,0 +1,54 @@
+Blob :: blob {
+    a: bool,
+    b: int,
+    c: float,
+}
+
+BlobBlob :: blob {
+    _blob: Blob,
+}
+
+start :: fn do
+    b1 := Blob {
+        a: true,
+        b: 1,
+        c: 0.,
+    }
+
+    b1_eq := Blob {
+        a: true,
+        b: 1,
+        c: 0.,
+    }
+
+    b2 := Blob {
+        a: false,
+        b: 420,
+        c: .0,
+    }
+
+    b1 <=> b1_eq
+    b1 != b2 <=> true
+
+    bb1 := BlobBlob {
+        _blob: b1,
+    }
+
+    bb1_eq := BlobBlob {
+        _blob: b1,
+    }
+
+    bb1 <=> bb1_eq
+
+    bb1_eq2 := BlobBlob {
+        _blob: b1_eq,
+    }
+
+    bb1 <=> bb1_eq2
+
+    bb2 := BlobBlob {
+        _blob: b2,
+    }
+
+    bb1 != bb2 <=> true
```

Comment:
It would be nice to see a negative test as well - so a case where they don't match - just to make sure we haven't made a tautology...

Reply:
I'm not sure that I follow, isn't the assertion on line 53 a negative test since it checks for inequality between `bb1` and `bb2`? Or do you want me to add `bb1 == bb2 <=> false`? I'm not the most well read on testing.

Reply:
I think it looks good. There are plenty of negative tests, maybe Edvard was just tired?

Reply:
I was tired....